src/devices/device_models/hydro_generation.jl

```diff
@@ -252,3 +450,103 @@ function cost_function(canonical::Canonical,
     return
 
 end
+
+##################################### Water/Energy Budget Constraint ############################
+function _get_budget(canonical::Canonical,
```

Comment:
This doesn't seem right, the budget is assumed to be constant at the end of the LookAhead. This is repetitive from the RunofRiver.

Reply:
@sourabhdalvi bump about this comment.